pallets/authors-manager/src/lib.rs

```diff
@@ -478,6 +517,349 @@ impl<T: Config> Pallet<T> {
         }
     }
 
+    fn validate_author_registration_request(
+        account_id: &T::AccountId,
+        eth_public_key: &ecdsa::Public,
+    ) -> DispatchResult {
+        let author_account_ids = Self::author_account_ids().ok_or(Error::<T>::NoAuthors)?;
+        ensure!(!author_account_ids.is_empty(), Error::<T>::NoAuthors);
+
+        ensure!(!author_account_ids.contains(account_id), Error::<T>::AuthorAlreadyExists);
+
+        ensure!(
+            !<EthereumPublicKeys<T>>::contains_key(eth_public_key),
+            Error::<T>::AuthorEthKeyAlreadyExists
+        );
+
+        ensure!(
+            author_account_ids.len() < (<MaximumAuthorsBound as sp_core::TypedGet>::get() as usize),
+            Error::<T>::MaximumAuthorsReached
+        );
+
+        let validator_id = <T as SessionConfig>::ValidatorIdOf::convert(account_id.clone())
+            .ok_or(Error::<T>::ErrorConvertingAccountIdToAuthorId)?;
+
+        ensure!(
+            <pallet_session::NextKeys<T>>::contains_key(&validator_id),
+            Error::<T>::AuthorSessionKeysNotSet
+        );
+
+        Ok(())
+    }
+
+    fn validate_author_deregistration_request(account_id: &T::AccountId) -> DispatchResult {
+        let author_account_ids = Self::author_account_ids().ok_or(Error::<T>::NoAuthors)?;
+
+        ensure!(
+            author_account_ids.len() > T::MinimumAuthorsCount::get() as usize,
+            Error::<T>::MinimumAuthorsReached
+        );
+
+        // Check if the author exists in the current list of authors
+        ensure!(author_account_ids.contains(account_id), Error::<T>::AuthorNotFound);
+
+        // Check if any author already has a deregistration in progress (global check)
+        ensure!(
+            !Self::has_any_active_deregistration(),
+            Error::<T>::DeregistrationAlreadyInProgress
+        );
+
+        // Check if this author has any active actions (registration or deregistration)
+        ensure!(!Self::has_any_active_action(account_id), Error::<T>::RemovalAlreadyRequested);
```

Comment:
Please see [this comment](https://github.com/AventusProtocolFoundation/avn-parachain/pull/588/files#r2504370831) on main